11_notify/kasunfdo/pkg/puppy/rest_test.go

```diff
@@ -0,0 +1,155 @@
+package puppy_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/anz-bank/go-course/11_notify/kasunfdo/pkg/puppy"
+	"github.com/anz-bank/go-course/11_notify/kasunfdo/pkg/puppy/store"
+	"github.com/go-chi/chi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func getTestRouter(t *testing.T) *chi.Mux {
+	storer := store.NewMapStore()
+	_, err := storer.CreatePuppy(puppy.Puppy{ID: 1, Breed: "Labrador", Colour: "Cream", Value: 2000})
+	require.NoError(t, err)
+
+	router := chi.NewRouter()
+	apiHandler := *puppy.NewAPIHandler(storer, "")
```

Comment:
Here also you are dereferencing the handler again, and I'm not sure why. You didn't do it in your lab 10 (I see this as a diff when I looked at the diff). Maybe I'm just sleep-deprived and I'm missing something here.

Reply:
Same reason as above. 😄